doc/source/whatsnew/v1.0.0.rst

```diff
@@ -38,7 +38,22 @@ Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 - :class:`pandas.core.groupby.GroupBy.transform` now raises on invalid operation names (:issue:`27489`).
--
+- :class:`pandas.core.arrays.IntervalArray` adopts a new ``__repr__`` in accordance with other array classes (:issue:`25022`)
+
+*pandas 0.25.x*
+
+.. code-block:: python
+
+   intv_arr = pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
+   intv_arr
+
+*pandas 1.0.0*
+
+.. code-block:: python
```

Comment:
In case it helps you should use `code-block` for the old behavior and the `ipython` directive for newly introduced behavior. You can see an example of this with the Better Repr for MultiIndex section back in the 0.25 whatsnew

https://github.com/pandas-dev/pandas/blob/5f31bca3813b2ed71363a9d40518f217d2b546d6/doc/source/whatsnew/v0.25.0.rst#better-repr-for-multiindex